dotnet/src/SemanticKernel.Core/Memory/Mem0/Mem0Client.cs

```diff
@@ -0,0 +1,182 @@
+
+        // Delete.
+        var responseMessage = await this._httpClient.DeleteAsync(clearMemoryUrl).ConfigureAwait(false);
+        responseMessage.EnsureSuccessStatusCode();
+    }
+
+    internal sealed class CreateMemoryRequest
+    {
+        [JsonPropertyName("app_id")]
+        public string? AppId { get; set; }
+        [JsonPropertyName("agent_id")]
+        public string? AgentId { get; set; }
+        [JsonPropertyName("run_id")]
+        public string? RunId { get; set; }
+        [JsonPropertyName("user_id")]
+        public string? UserId { get; set; }
+        [JsonPropertyName("messages")]
+        public CreateMemoryMessage[] Messages { get; set; } = [];
+    }
+
+    internal sealed class CreateMemoryMessage
+    {
+        [JsonPropertyName("content")]
+        public string Content { get; set; } = string.Empty;
+        [JsonPropertyName("role")]
+        public string Role { get; set; } = string.Empty;
+    }
+
+    internal sealed class SearchRequest
+    {
+        [JsonPropertyName("app_id")]
+        public string? AppId { get; set; }
+        [JsonPropertyName("agent_id")]
+        public string? AgentId { get; set; } = null;
+        [JsonPropertyName("run_id")]
+        public string? RunId { get; set; } = null;
+        [JsonPropertyName("user_id")]
+        public string? UserId { get; set; } = null;
+        [JsonPropertyName("query")]
+        public string Query { get; set; } = string.Empty;
+    }
+
+    internal sealed class SearchResponseItem
+    {
+        [JsonPropertyName("id")]
+        public string Id { get; set; } = string.Empty;
+        [JsonPropertyName("memory")]
+        public string Memory { get; set; } = string.Empty;
+        [JsonPropertyName("hash")]
+        public string Hash { get; set; } = string.Empty;
+        [JsonPropertyName("metadata")]
+        public object? Metadata { get; set; }
+        [JsonPropertyName("score")]
+        public double Score { get; set; }
+        [JsonPropertyName("created_at")]
+        public DateTime CreatedAt { get; set; }
+        [JsonPropertyName("updated_at")]
+        public DateTime? UpdatedAt { get; set; }
+        [JsonPropertyName("user_id")]
+        public string UserId { get; set; } = string.Empty;
+        [JsonPropertyName("app_id")]
+        public string? AppId { get; set; }
+        [JsonPropertyName("agent_id")]
+        public string AgentId { get; set; } = string.Empty;
+        [JsonPropertyName("session_id")]
+        public string RunId { get; set; } = string.Empty;
+    }
+}
+
+[JsonSourceGenerationOptions(JsonSerializerDefaults.General,
+    UseStringEnumConverter = false,
+    DefaultIgnoreCondition = JsonIgnoreCondition.WhenWritingNull,
+    WriteIndented = false)]
+[JsonSerializable(typeof(Mem0Client.CreateMemoryRequest))]
+[JsonSerializable(typeof(Mem0Client.SearchRequest))]
+[JsonSerializable(typeof(Mem0Client.SearchResponseItem[]))]
+internal partial class Mem0SourceGenerationContext : JsonSerializerContext
+{
+}
```

Comment:
Move as internal class (to keep in the same file)
```suggestion
[JsonSourceGenerationOptions(JsonSerializerDefaults.General,
    UseStringEnumConverter = false,
    DefaultIgnoreCondition = JsonIgnoreCondition.WhenWritingNull,
    WriteIndented = false)]
[JsonSerializable(typeof(Mem0Client.CreateMemoryRequest))]
[JsonSerializable(typeof(Mem0Client.SearchRequest))]
[JsonSerializable(typeof(Mem0Client.SearchResponseItem[]))]
internal partial class Mem0SourceGenerationContext : JsonSerializerContext
{
}
```

Reply:
These cannot be moved to an internal class, since it breaks the source generation.